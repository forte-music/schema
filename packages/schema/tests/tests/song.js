import gql from 'graphql-tag';
import { print } from 'graphql/language/printer';

import SongFields from './fragments/SongFields.graphql';
import AlbumFields from './fragments/AlbumFields.graphql';
import ArtistFields from './fragments/ArtistFields.graphql';
import UserStatsFields from './fragments/UserStatsFields.graphql';
import SongUserStatsFields from './fragments/SongUserStatsFields.graphql';

import client from '../client';

const variables = { songId: 'song:2:1' };
it('should get a song by id', async () => {
  const query = gql`
    query($songId: ID!) {
      song(id: $songId) {
        ...SongFields

        album {
          ...AlbumFields
        }

        artists {
          ...ArtistFields
        }

        stats {
          ...UserStatsFields
        }

        songStats {
          ...SongUserStatsFields
        }
      }
    }

    ${SongFields}
    ${AlbumFields}
    ${ArtistFields}
    ${UserStatsFields}
    ${SongUserStatsFields}
  `;

  const data = await client.request(print(query), variables);

  expect(data).toMatchSnapshot();
});

it('should get all items when limit is -1', async () => {
  const { songs } = await client.request(
    `
      query {
        songs(first: -1) {
          pageInfo {
            hasNextPage
            count
          }
          edges {
            cursor
            node {
              id
            }
          }
        }
      }
    `
  );

  expect(songs.pageInfo.count).toBeTruthy();
  expect(songs.pageInfo.count).toEqual(songs.edges.length);
});
