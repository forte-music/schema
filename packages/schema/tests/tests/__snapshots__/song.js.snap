// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get a song by id 1`] = `
Object {
  "song": Object {
    "album": Object {
      "artworkUrl": "http://is4.mzstatic.com/image/thumb/Music5/v4/08/da/96/08da9619-3f9b-7c95-60d1-6c18cfdd4dbd/source/600x600bb.jpg",
      "duration": 211,
      "id": "00000000000000000000000000000002",
      "name": "I'm That (Remix)",
      "releaseYear": 2015,
    },
    "artists": Array [
      Object {
        "id": "00000000000000000000000000000002",
        "name": "R. City",
      },
      Object {
        "id": "00000000000000000000000000000015",
        "name": "Beenie Man",
      },
      Object {
        "id": "00000000000000000000000000000016",
        "name": "Azealia Banks",
      },
    ],
    "id": "00000000000000000000000000000002",
    "name": "I'm That (Remix)",
    "songStats": Object {
      "id": "00000000000000000000000000000002:stats:song",
      "liked": false,
      "playCount": 0,
    },
    "stats": Object {
      "id": "00000000000000000000000000000002:stats",
      "lastPlayed": null,
    },
    "streamUrl": "/music/track.mp3",
    "trackNumber": 1,
  },
}
`;
